packages/agreement-process/test/getAgreements.test.ts

```diff
@@ -560,4 +614,20 @@ describe("get agreements", () => {
       results: expect.arrayContaining([agreement5, agreement6, agreement7]),
     });
   });
+
+  it("should get agreements for a delegated eservice with filters: consumerId and requester is consumer", async () => {
+    const agreements = await agreementService.getAgreements(
+      {
+        consumerId: tenant4.id,
+      },
+      tenant4.id,
+      10,
+      0,
+      genericLogger
+    );
+    expect(agreements).toEqual({
+      totalCount: 1,
+      results: expect.arrayContaining([agreement8]),
+    });
+  });
```

Comment:
Since we can have multiple consumer delegations active for the same eservice, could you add some more consumer delegations with their related agreements and make sure only the ones related to the caller-delegate are returned? Also, it could be good to test that the returned value contains both my own agreements and the ones for which I am consumer delegate (maybe adding another test for that)

Reply:
2de45d1a8c036cdefcc668dba3506fe9d9aabf33